### PR TITLE
Change the C++ code to use the new DegreesOfFreedom wherever possible

### PR DESCRIPTION
**Pull request by pleroy**
C# not affected for now.

### CODE REVIEW

File: ksp_plugin/physics_bubble.cpp [hunk 1]
```diff
@@ -184,23 +184,13 @@ std::vector<Vessel*> PhysicsBubble::vessels() const {
...
-Velocity<Barycentric> const&
-PhysicsBubble::velocity_from_centre_of_mass(
-    Vessel const* const vessel) const {
-  CHECK(!empty()) << "Empty bubble";
-  CHECK(current_->velocities_from_centre_of_mass != nullptr);
-  auto const it = current_->velocities_from_centre_of_mass->find(vessel);
-  CHECK(it != current_->velocities_from_centre_of_mass->end());
+  CHECK(current_->from_centre_of_mass != nullptr);
+  auto const it = 
+      current_->from_centre_of_mass->find(vessel);
```
Comment (eggrobin) [inline ksp_plugin/physics_bubble.cpp]: ?

File: ksp_plugin/physics_bubble.cpp [hunk 2]
```diff
@@ -261,24 +248,13 @@ void PhysicsBubble::ComputeNextVesselOffsets(
...
-    next->displacements_from_centre_of_mass->emplace(
-        vessel,
-        displacement_from_centre_of_mass);
-    next->velocities_from_centre_of_mass->emplace(
-        vessel,
-        velocity_from_centre_of_mass);
+                vessel_degrees_of_freedom - *next->centre_of_mass));
+    VLOG(1) << NAMED(from_centre_of_mass);
+    next->from_centre_of_mass->emplace(
+        vessel, from_centre_of_mass);
```
Comment (eggrobin) [inline ksp_plugin/physics_bubble.cpp]: ?

File: ksp_plugin/plugin.cpp [hunk 3]
```diff
@@ -317,25 +309,18 @@ void Plugin::InsertCelestial(
       std::make_unique<Celestial>(
           std::make_unique<MassiveBody>(gravitational_parameter)));
   CHECK(inserted.second) << "Body already exists at index " << celestial_index;
-  LOG(INFO) << "Initial |orbit.pos| for celestial at index " << celestial_index
-            << ": " << from_parent_position;
-  Displacement<Barycentric> const displacement =
+  LOG(INFO) << "Initial {|orbit.pos|, |orbit.vel|} for celestial at index "
```
Comment (eggrobin) [inline ksp_plugin/plugin.cpp]: move pipes

File: ksp_plugin/plugin.cpp [hunk 4]
```diff
@@ -434,61 +409,25 @@ void Plugin::AdvanceTime(Instant const& t, Angle const& planetarium_rotation) {
...
-  CHECK(vessel->is_initialized()) << "Vessel with GUID " << vessel_guid
-                                  << " was not given an initial state";
-  Velocity<Barycentric> const barycentric_result =
-      vessel->prolongation().last().degrees_of_freedom().velocity() -
-      vessel->parent().prolongation().last().degrees_of_freedom().velocity();
-  Velocity<AliceSun> const result =
+  auto const barycentric_result =
+      vessel->prolongation().last().degrees_of_freedom() -
+      vessel->parent().prolongation().last().degrees_of_freedom();
+  auto const result =
```
Comment (eggrobin) [inline ksp_plugin/plugin.cpp]: ?

File: ksp_plugin/plugin.cpp [hunk 5]
```diff
@@ -496,13 +435,13 @@ Velocity<AliceSun> Plugin::CelestialParentRelativeVelocity(
...
   CHECK(celestial.has_parent())
       << "Body at index " << celestial_index << " is the sun";
-  Velocity<Barycentric> const barycentric_result =
-      celestial.prolongation().last().degrees_of_freedom().velocity() -
-      celestial.parent().prolongation().last().degrees_of_freedom().velocity();
-  Velocity<AliceSun> const result =
+  auto const barycentric_result =
+      celestial.prolongation().last().degrees_of_freedom() -
+      celestial.parent().prolongation().last().degrees_of_freedom();
+  auto const result =
```
Comment (eggrobin) [inline ksp_plugin/plugin.cpp]: ?

File: ksp_plugin/plugin.hpp [hunk 6]
```diff
@@ -156,36 +152,22 @@ class Plugin {
   // degrees.
   virtual void AdvanceTime(Instant const& t, Angle const& planetarium_rotation);
 
-  // Returns the position of the vessel with GUID |vessel_guid| relative to its
-  // parent at current time. For a KSP |Vessel| |v|, the argument corresponds to
-  // |v.id.ToString()|, the return value to |v.orbit.pos|.
+  // Returns the displacement and velocity of the vessel with GUID |vessel_guid|
+  // relative to its parent at current time. For a KSP |Vessel| |v|, the
+  // argument corresponds to  |v.id.ToString()|, the return value to
+  // {|v.orbit.pos|, |v.orbit.vel|}.
```
Comment (eggrobin) [inline ksp_plugin/plugin.hpp]: ?

File: ksp_plugin/plugin.hpp [hunk 7]
```diff
@@ -156,36 +152,22 @@ class Plugin {
...
-  virtual Displacement<AliceSun> CelestialDisplacementFromParent(
-      Index const celestial_index) const;
-
-  // Returns the velocity of the celestial at index |celestial_index| relative
-  // to its parent at current time. For a KSP |CelestialBody| |b|, the argument
-  // corresponds to |b.flightGlobalsIndex|, the return value to |b.orbit.vel|.
+  // Returns the displacement and velocity of the celestial at index
+  // |celestial_index| relative to its parent at current time. For a KSP
+  // |CelestialBody| |b|, the argument corresponds to |b.flightGlobalsIndex|,
+  // the return value to {|b.orbit.pos|, |b.orbit.vel|}.
```
Comment (eggrobin) [inline ksp_plugin/plugin.hpp]: ?

File: ksp_plugin/plugin.hpp [hunk 8]
```diff
@@ -134,12 +132,10 @@ class Plugin {
   // be called once per vessel. Must be called after initialization.
   // For a KSP |Vessel| |v|, the arguments correspond to
   // |v.id.ToString()|,
-  // |v.orbit.pos|,
-  // |v.orbit.vel|.
+  // {|v.orbit.pos|, |v.orbit.vel|}.
```
Comment (eggrobin) [inline ksp_plugin/plugin.hpp]: ?

File: testing_utilities/solar_system_test.cpp [hunk 9]
```diff
@@ -38,12 +40,13 @@ class SolarSystemTest : public testing::Test {
...
-    Velocity<ICRFJ2000Ecliptic> const v =
-        primary.last().degrees_of_freedom().velocity() -
-        secondary.last().degrees_of_freedom().velocity();
+    RelativeDegreesOfFreedom<ICRFJ2000Ecliptic> const primary_secondary =
+        primary.last().degrees_of_freedom() -
+        secondary.last().degrees_of_freedom();
+    Vector<Length, ICRFJ2000Ecliptic> const& r =
+        primary_secondary.displacement();
+    Velocity<ICRFJ2000Ecliptic> const& v =
+        primary_secondary.velocity();
```
Comment (eggrobin) [inline testing_utilities/solar_system_test.cpp]: ?

File: ksp_plugin/physics_bubble.cpp (hunk 1 above)
Comment (pleroy) [inline ksp_plugin/physics_bubble.cpp]: Done.

File: ksp_plugin/physics_bubble.cpp (hunk 2 above)
Comment (pleroy) [inline ksp_plugin/physics_bubble.cpp]: Done.

File: ksp_plugin/plugin.cpp (hunk 3 above)
Comment (pleroy) [inline ksp_plugin/plugin.cpp]: Done.

File: ksp_plugin/plugin.cpp (hunk 4 above)
Comment (pleroy) [inline ksp_plugin/plugin.cpp]: Done.

File: ksp_plugin/plugin.cpp (hunk 5 above)
Comment (pleroy) [inline ksp_plugin/plugin.cpp]: Done.

File: ksp_plugin/plugin.hpp (hunk 6 above)
Comment (pleroy) [inline ksp_plugin/plugin.hpp]: Done.

File: ksp_plugin/plugin.hpp (hunk 7 above)
Comment (pleroy) [inline ksp_plugin/plugin.hpp]: Done.

File: ksp_plugin/plugin.hpp (hunk 8 above)
Comment (pleroy) [inline ksp_plugin/plugin.hpp]: Done.

File: testing_utilities/solar_system_test.cpp (hunk 9 above)
Comment (pleroy) [inline testing_utilities/solar_system_test.cpp]: Done.

**Pull request merged**